op-monitorism/faultproof_withdrawals/validator/l2_proxy.go

```diff
@@ -0,0 +1,185 @@
+package validator
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+type L2Proxy struct {
+	l2GethClient          *ethclient.Client
+	chainID               *big.Int
+	ctx                   context.Context
+	l2OpGethBackupClients map[string]*ethclient.Client
+	ConnectionError       map[string]uint64
+	Connections           map[string]uint64
+}
+
```

Comment:
but the `connectionsErrors` is not the same type with the `L1Proxy`
Here, the type `L2Proxy` has similar type + name with the `L1Proxy` I think we should make them closer to avoid confusion in the future. This means that this type would be a generic type called `proxy` and we instantiate a L1Proxy from proxy and l2proxy from proxy with the specification.